middle_end/flambda/compilenv_deps/target_imm.ml

```diff
@@ -5,243 +5,289 @@
 (*                       Pierre Chambart, OCamlPro                        *)
 (*           Mark Shinwell and Leo White, Jane Street Europe              *)
 (*                                                                        *)
-(*   Copyright 2013--2017 OCamlPro SAS                                    *)
-(*   Copyright 2014--2017 Jane Street Group LLC                           *)
+(*   Copyright 2013--2021 OCamlPro SAS                                    *)
+(*   Copyright 2014--2021 Jane Street Group LLC                           *)
 (*                                                                        *)
 (*   All rights reserved.  This file is distributed under the terms of    *)
 (*   the GNU Lesser General Public License version 2.1, with the          *)
 (*   special exception on linking described in the file LICENSE.          *)
 (*                                                                        *)
 (**************************************************************************)
 
-[@@@ocaml.warning "+a-4-30-40-41-42"]
+[@@@ocaml.warning "+a-30-40-41-42"]
 
-module TO = Targetint.OCaml
+let () =
+  match Targetint.num_bits with
+  | Sixty_four -> ()
+  | Thirty_two ->
+    if Config.flambda then begin
+      Misc.fatal_error "Flambda 2 does not yet support 32-bit compilation"
+    end
+
+module Imm = struct
+  module T0 = struct
+    include Numbers.Int64
+
+    let compare = compare
```

Comment:
Would it be worth it to ensure that these functions are specialized to `Int64.t` ?

Reply:
CR added